core/src/commonMain/kotlin/com/ustadmobile/core/controller/SchoolDetailPresenter.kt

```diff
@@ -1,60 +1,49 @@
+import com.ustadmobile.door.DoorLiveData
 import com.ustadmobile.lib.db.entities.School
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.Runnable
-import kotlinx.coroutines.launch
+import com.ustadmobile.lib.db.entities.UmAccount
+import kotlinx.coroutines.withTimeoutOrNull
+
 
-/**
- *  Presenter for SchoolDetail view
- **/
 class SchoolDetailPresenter(context: Any,
-                            arguments: Map<String, String>?,
-                            view: SchoolDetailView,
-                            val impl: UstadMobileSystemImpl = UstadMobileSystemImpl.instance,
-                            val repository: UmAppDatabase =
-                                    UmAccountManager.getRepositoryForActiveAccount(context))
-    : UstadBaseController<SchoolDetailView>(context, arguments!!, view) {
+                          arguments: Map<String, String>, view: SchoolDetailView,
+                          lifecycleOwner: DoorLifecycleOwner,
+                          systemImpl: UstadMobileSystemImpl,
+                          db: UmAppDatabase, repo: UmAppDatabase,
+                          activeAccount: DoorLiveData<UmAccount?> = UmAccountManager.activeAccountLiveData)
+    : UstadDetailPresenter<SchoolDetailView, School>(context, arguments, view, lifecycleOwner, systemImpl,
+        db, repo, activeAccount) {
+
+    override val persistenceMode: PersistenceMode
+        get() = PersistenceMode.DB
 
-    private var schoolDao: SchoolDao = repository.schoolDao
-    private lateinit var currentSchool : School
-    private lateinit var factory: DataSource.Factory<Int, School>
 
     override fun onCreate(savedState: Map<String, String>?) {
         super.onCreate(savedState)
 
-        if(arguments.containsKey(ARG_SCHOOL_DETAIL_SCHOOL_UID)){
-            GlobalScope.launch {
-                val school =
-                        schoolDao.findByUidAsync(arguments[ARG_SCHOOL_DETAIL_SCHOOL_UID]?.toLong() ?: 0)
-                if (school != null) {
-                    currentSchool = school
-                    view.runOnUiThread(Runnable {
-                        view.setSchool(school)
-                        view.setUpTabs(listOf(ClazzListView.VIEW_NAME,
-                                ClazzStudentListView.VIEW_NAME))
-                    })
-                }
-            }
-        }
     }
 
-    /**
-     * Handle what happens when the gear edit button is clicked - should go to schooledit page
-     */
-    fun handleClickSchoolEdit(){
-        val args = mapOf(ARG_SCHOOL_DETAIL_SCHOOL_UID to currentSchool.schoolUid.toString())
-        impl.go(SchoolEditView.VIEW_NAME, args, context)
+    override suspend fun onLoadEntityFromDb(db: UmAppDatabase): School? {
+        val entityUid = arguments[ARG_ENTITY_UID]?.toLong() ?: 0L
+
+        val school =  withTimeoutOrNull(2000) {
+            db.schoolDao.findByUidAsync(entityUid)
+        } ?: School()
+
+        view.setTitle(school.schoolName?:"")
```

Comment:
I think to keep things consistent it would be better to use an abstract var with set/get methods on the view instead of having a method. E.g. then we would have view.title = school?.schoolName ?: ""